modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/maven/AbstractMavenDeployableProjectModelFactory.java

```diff
@@ -0,0 +1,453 @@
+                             .setGroupId(d.getArtifactCoordinates().getGroupId())
+                             .setClassifier(d.getArtifactCoordinates().getClassifier())
+                             .setType(d.getArtifactCoordinates().getType())
+                             .setVersion(d.getArtifactCoordinates().getVersion())
+                             .setBaseVersion(d.getArtifactCoordinates().getVersion())
+                             .build())
+          .setBundleUri(bundle)
+          .build();
+    };
+  }
+
+  private File getPomFromFolder(File projectFolder) {
+    String pomFilePath = "pom.xml";
+
+    File pomFile = new File(projectFolder, pomFilePath);
+    checkState(pomFile.exists(),
+               format("The pom.xml file for artifact in folder %s could not be found",
+                      projectFolder.getAbsolutePath()));
+
+    return pomFile;
+  }
+
+  protected Optional<org.apache.maven.model.Plugin> findArtifactPackagerPlugin(Model model, List<String> activeProfiles) {
+    Stream<org.apache.maven.model.Plugin> basePlugin = Stream.empty();
+    Build build = model.getBuild();
+    if (build != null) {
+      basePlugin = findArtifactPackagerPlugin(build.getPlugins()).map(Stream::of).orElse(Stream.empty());
+    }
+
+    // Sort them so the processing is consistent with how Maven calculates the plugin configuration for the effective pom.
+    final List<String> sortedActiveProfiles = activeProfiles
+        .stream()
+        .sorted(String::compareTo)
+        .collect(toList());
+
+    final Stream<org.apache.maven.model.Plugin> packagerConfigsForActivePluginsStream = model.getProfiles().stream()
+        .filter(profile -> sortedActiveProfiles.contains(profile.getId()))
+        .map(profile -> findArtifactPackagerPlugin(profile.getBuild() != null ? profile.getBuild().getPlugins() : null))
+        .filter(plugin -> !plugin.equals(empty()))
+        .map(Optional::get);
+
+    return concat(basePlugin, packagerConfigsForActivePluginsStream)
+        .reduce((p1, p2) -> {
+          p1.setConfiguration(mergeXpp3Dom((Xpp3Dom) p2.getConfiguration(), (Xpp3Dom) p1.getConfiguration()));
+          p1.getDependencies().addAll(p2.getDependencies());
+
+          return p1;
+        });
+  }
+
+  private Optional<org.apache.maven.model.Plugin> findArtifactPackagerPlugin(List<org.apache.maven.model.Plugin> plugins) {
+    if (plugins != null) {
+      return plugins.stream().filter(plugin -> (plugin.getArtifactId().equals(MULE_MAVEN_PLUGIN_ARTIFACT_ID)
+          && plugin.getGroupId().equals(MULE_MAVEN_PLUGIN_GROUP_ID)) ||
+          (plugin.getArtifactId().equals(MULE_EXTENSIONS_PLUGIN_ARTIFACT_ID) &&
+              plugin.getGroupId().equals(MULE_EXTENSIONS_PLUGIN_GROUP_ID)))
+          .findFirst();
+    } else {
+      return empty();
+    }
+  }
+
+  private List<org.mule.runtime.module.artifact.activation.internal.plugin.Plugin> getAdditionalPluginDependencies(org.apache.maven.model.Plugin packagingPlugin) {
```

Comment:
doesn't this have a lot in common with wat `AdditionalPluginDependenciesResolver` also does?

Reply:
In `AbstractMavenDeployableProjectModelFactory` we're parsing the additional plugin dependencies configured for the application, whereas `AdditionalPluginDependenciesResolver` performs some checks on the additional plugin dependencies of each plugin present in the application.